x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/terms/index.tsx

```diff
@@ -50,6 +58,100 @@ function isSortableByColumn(layer: IndexPatternLayer, columnId: string) {
   );
 }
 
+function getDisallowedTermsMessage(
+  layer: IndexPatternLayer,
+  columnId: string,
+  indexPattern: IndexPattern
+) {
+  const hasMultipleShifts =
+    uniq(
+      Object.values(layer.columns)
+        .filter((col) => operationDefinitionMap[col.operationType].shiftable)
+        .map((col) => col.timeShift || '')
+    ).length > 1;
+  if (!hasMultipleShifts) {
+    return undefined;
+  }
+  return {
+    message: i18n.translate('xpack.lens.indexPattern.termsWithMultipleShifts', {
+      defaultMessage:
+        "Can't use multiple time shifts in a single layer together with dynamic top values. Either use the same time shift for all metrics or use filters instead of top values.",
```

Comment:
@flash1293 thank you for the clarification! 

To make the first sentence a full sentence, I would change it to:
`In a single layer, you are unable to combine multiple time shifts and dynamic top values.`

For the second sentence, I would simplify to:
`Use the same value for all time shifts, or use filters instead of top values.`

It is confusing that `Pin current top values` doesn't appear in the error message. Can we change it to:
`Use filters instead of top values` or `Use filters`?